src/hooks/useCountdownTimer/useCountdownTimer.tsx

```diff
@@ -0,0 +1,27 @@
+import { useCallback, useEffect, useState } from 'react'
+
+export const useCountdownTimer = (initialSeconds: number) => {
+  const [secondsRemaining, setSecondsRemaining] = useState(initialSeconds)
+  const [progress, setProgress] = useState(100)
+
+  const reset = useCallback(() => {
+    setSecondsRemaining(initialSeconds)
+    setProgress(100)
+  }, [initialSeconds])
+
+  useEffect(() => {
+    const intervalId = setInterval(() => {
```

Comment:
if an interval exists, will need to clear the interval before creating a new one